implementations/code_df/code_changes_lines.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime
+
+import conditions
+import utils
+from commit import Commit
+
+
+class CodeChangesLines(Commit):
+    """
+    Class for Code_Changes_Lines
+
+    :param items: A list of dictionaries, each element a line from the
+        JSON file with Perceval data
+
+    :param date_range: A tuple which represents the start and end date of
+        interest
+
+    :param is_code:  list of CodeCondition objects
+        It is used to determine what comprises source code.
+
+    :param conds: A list of Commit objects. It is used to
+        exclude or include different kinds of commits, like
+        empty or merge commits, or commits made on the master
+        branch.
+    """
+
+    def __init__(self, items, date_range=(None, None),
```

Comment:
Isn't __init__ in the Commit class now?

Reply:
So the problem here is that unlike the other metrics we have implemented, this one cannot be directly computed by `sum` or `count` operations. I called the `__init__` method here so that I can add columns for modifications to the dataframe.
We can go about this in two ways:
- **Not add a column**  
Here would have to iterate over each file in every commit and add the number of lines modified to a "count" variable.
That would mean be a little messy, especially when it comes to aggregating over it.

- **Adding a column** 
This would allow for easier computation of the metric, and it is easier to understand for someone studying the code. The only problem, as of now, is having to add the column in the `__init__ ` method, which I have done here.

Reply:
Now I understand. Thanks for the clarification.

Consider the following approach: you can redefine _flatten, and for each commit, you can compute the number of lines added and removed in all files, and get from them the total number of lines modified, and build that directly into the flattened dictionary (a new column, "modified", with the number of lines modified).

Then, you can write a `_agg` file that uses `sum()` for the aggregation.

I think that would be cleaner... (and maybe  reason for having _agg at the level of the specific metric, instead than at the Commit class level, but we will see...)

Reply:
That would mean modifying the `_flatten` method in the `Commit` class. Is it a good idea to modify a parent class?

Reply:
Let's see when it is implemented, but methods in a class hierarchy are written to be overridden. That's the idea of subclassing ;-)

Reply:
Yes @jgbarah, my only problem is that this won't be the case for all metrics, meaning that all implementation would not have a similar structure (not all of them would have an overridden _flatten method).

Anyways, thanks for approving it.